Remove Element::userAgentPartId() in favor of pseudo()

annevk opened the pull request:
#### a9f625b0b81f3e6dd73f16c2504a40f58fcec8cf
<pre>
Remove Element::userAgentPartId() in favor of pseudo()
<a href="https://bugs.webkit.org/show_bug.cgi?id=267030">https://bugs.webkit.org/show_bug.cgi?id=267030</a>

Reviewed by Tim Nguyen.

This virtual method was never overridden and didn&apos;t do anything beyond
calling pseudo() directly.

Also remove an obsolete include in YouTubeEmbedShadowElement and use
pseudo() instead of directly invoking the attribute getter in
TextFieldInputType.

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::userAgentPartId const): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/html/shadow/DateTimeEditElement.cpp:
(WebCore::DateTimeEditElement::layout):
* Source/WebCore/html/shadow/YouTubeEmbedShadowElement.cpp:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::autoAppearanceForElement const):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::matchPartPseudoElementRules):
(WebCore::Style::ElementRuleCollector::collectMatchingShadowPseudoElementRules):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/style/StyleInvalidator.cpp:
(WebCore::Style::Invalidator::invalidateShadowPseudoElements):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::elementTypeHasAppearanceFromUAStyle):
* Source/WebCore/style/StyleSharingResolver.cpp:
(WebCore::Style::SharingResolver::canShareStyleWithElement const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::userAgentPartId):

Canonical link: <a href="https://commits.webkit.org/272604@main">https://commits.webkit.org/272604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/603e9300f6b4753d5aaa4b5f75a516cb58410403

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32333 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11070 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34155 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29270 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33183 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13423 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8276 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28819 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32751 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9336 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28896 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/8131 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8269 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28827 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36228 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29397 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29256 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/34400 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8398 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6345 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32263 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10057 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7538 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9028 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8962 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/style/StyleInvalidator.cpp:
```diff
@@ -421,7 +421,7 @@ void Invalidator::invalidateShadowPseudoElements(ShadowRoot& shadowRoot)
 
     for (auto& descendant : descendantsOfType<Element>(shadowRoot)) {
-        auto& partId = descendant.userAgentPartId();
+        auto& partId = descendant.pseudo();
         if (!partId)
```
You could rename these variables to `pseudo` or something.

annevk replied:
I think we might want to rename pseudo to userAgentPart at some point. I suppose I could try to tackle that here...

annevk replied:
I think I prefer leaving this to a removal of the wrapper and then we can rename all the things in a follow-up.

nt1m replied:
Yeah that would make sense.

nt1m commented on Source/WebCore/style/ElementRuleCollector.cpp:
```diff
@@ -402,7 +402,7 @@ void ElementRuleCollector::collectMatchingShadowPseudoElementRules(const MatchRe
         collectMatchingRulesForList(&rules.cuePseudoRules(), matchRequest);
 #endif
-    auto& partId = element().userAgentPartId();
+    auto& partId = element().pseudo();
     if (!partId.isEmpty())
         collectMatchingRulesForList(rules.shadowPseudoElementRules(partId), matchRequest);
```
You could rename this var to `pseudo`

nt1m commented on Source/WebCore/rendering/RenderTheme.cpp:
```diff
@@ -362,7 +362,7 @@ StyleAppearance RenderTheme::autoAppearanceForElement(RenderStyle& style, const
 
     if (element->isInUserAgentShadowTree()) {
-        auto& part = element->userAgentPartId();
+        auto& part = element->pseudo();
```
You can rename this var to `pseudo`

nt1m commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -1425,7 +1425,7 @@ ExceptionOr<String> Internals::shadowRootType(const Node& root) const
 
 const AtomString& Internals::userAgentPartId(Element& element)
```
Can we rename this method and its setter for consistency to pseudo() and setPseudo()?

nt1m replied:
It's just a mass rename that needs to be done in LayoutTests/, shouldn't be too hard.

nt1m commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -3124,11 +3124,6 @@ CustomElementDefaultARIA* Element::customElementDefaultARIAIfExists()
 
-const AtomString& Element::userAgentPartId() const
-{
-    return pseudo();
```
This method might need WEBCORE_EXPORT to build

annevk committed:
Remove Element::userAgentPartId() in favor of pseudo()
https://bugs.webkit.org/show_bug.cgi?id=267030

Reviewed by Tim Nguyen.

This virtual method was never overridden and didn't do anything beyond
calling pseudo() directly.

Also remove an obsolete include in YouTubeEmbedShadowElement and use
pseudo() instead of directly invoking the attribute getter in
TextFieldInputType.

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::userAgentPartId const): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/html/shadow/DateTimeEditElement.cpp:
(WebCore::DateTimeEditElement::layout):
* Source/WebCore/html/shadow/YouTubeEmbedShadowElement.cpp:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::autoAppearanceForElement const):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::matchPartPseudoElementRules):
(WebCore::Style::ElementRuleCollector::collectMatchingShadowPseudoElementRules):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/style/StyleInvalidator.cpp:
(WebCore::Style::Invalidator::invalidateShadowPseudoElements):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::elementTypeHasAppearanceFromUAStyle):
* Source/WebCore/style/StyleSharingResolver.cpp:
(WebCore::Style::SharingResolver::canShareStyleWithElement const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::userAgentPartId):

Canonical link: https://commits.webkit.org/272604@main
